### PR TITLE
Feature/descw 1902 database deployment

### PR DESCRIPTION
**Pull request by mhaswell-bcgov**

### CODE REVIEW

File: deployments/kustomize/base/app/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+configMapGenerator:
+  - name: config
+    literals:
+    - NAAD_NAME=NAAD-1
+    - NAAD_URL=streaming1.naad-adna.pelmorex.com
+    - DESTINATION_URL=https://localhost/wp-json/naad/v1/alert
+    - DESTINATION_USER=naadbot
+
+secretGenerator:
+  - name: naad-secrets
```
Comment (ShawnTurple) [inline deployments/kustomize/base/app/kustomization.yaml:19]: does the word secrets need to be in the secret, as the type is a secret, so i suspect is redundant

File: deployments/kustomize/base/database/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: mariadb-deployment
```
Comment (ShawnTurple) [inline deployments/kustomize/base/database/deployment.yaml:4]: is this a best practice to put deployment at the end of considering it is a deployment... If you give me a clear indication why we need to do this, otherwise i believe we could remove deployment.

File: deployments/kustomize/base/database/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+configMapGenerator:
+  - name: database-config
+    literals:
+    - MARIADB_ALLOW_EMPTY_ROOT_PASSWORD="0" # If it is 1 and root_password is set, root_password takes precedence.
+
+secretGenerator:
+  - name: database-secrets
+    type: Opaque
+    literals:
+    - MARIADB_ROOT_PASSWORD="$z8ez=^)|YPPD]9NsJ*!.IWOpD.lgK(yH\k9J777g8(i3OJW["
```
Comment (ShawnTurple) [inline deployments/kustomize/base/database/kustomization.yaml:22]: since this password would only be used for local, is there a way to update it, so it doesn't look like a production password or forgot to change it.

File: deployments/kustomize/base/kustomization.yaml [hunk 4]
```diff
@@ -1,22 +1,8 @@
...
-    - DESTINATION_USER=naadbot
-
-secretGenerator:
-  - name: secrets
-    type: Opaque
-    literals:
-    - DESTINATION_PASSWORD='AAAA AAAA AAAA AAAA'
\ No newline at end of file
+# Overrides the namespace
+namespace: default
```
Comment (ShawnTurple) [inline deployments/kustomize/base/kustomization.yaml:8]: why did you remove the commonLabels?

File: src/database_connection_check.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+require_once 'vendor/autoload.php';
+
+use Doctrine\DBAL\DriverManager;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\ORMSetup;
+
+// Create a simple "default" Doctrine ORM configuration.
+$config = ORMSetup::createAttributeMetadataConfiguration(
+    paths: [__DIR__ . '/src'],
+    isDevMode: true,
```
Comment (ShawnTurple) [inline src/database_connection_check.php:11]: i'm assuming later you are going to have an env to change this?

**Review by ShawnTurple — DISMISSED** [review]

**Review by ShawnTurple — DISMISSED** [review]

File: deployments/kustomize/base/database/deployment.yaml (hunk 2 above)
Comment (mhaswell-bcgov) [reply deployments/kustomize/base/database/deployment.yaml]: Removed all of these names that reference their own resource type that I think I can.

File: deployments/kustomize/base/database/kustomization.yaml (hunk 3 above)
Comment (mhaswell-bcgov) [reply deployments/kustomize/base/database/kustomization.yaml]: Changed to `rootpassword` so there's no confusion.

File: deployments/kustomize/base/kustomization.yaml (hunk 4 above)
Comment (mhaswell-bcgov) [reply deployments/kustomize/base/kustomization.yaml:12]: Added them back although I don't fully understand what they're doing.

File: src/database_connection_check.php (hunk 5 above)
Comment (mhaswell-bcgov) [reply src/database_connection_check.php:11]: Yeah that would make sense. I'll have to look into what isDevMode actually does. I just took this from the Doctrine docs.

**Review by ShawnTurple — APPROVED** [review]
